packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -278,10 +302,29 @@ const TimePicker: TimePickerComponent = React.forwardRef<
             {...rest}
             {...readOnlyProps}
           />
+          {error ? (
+            <div className={`${prefix}--time-picker__error__icon`}>
+              {invalid ? (
+                <WarningFilled
+                  className={`${prefix}--checkbox__invalid-icon`}
+                  size={16}
+                />
+              ) : (
+                <WarningAltFilled
+                  className={`${prefix}--text-input__invalid-icon--warning`}
+                  size={16}
+                />
+              )}
+            </div>
+          ) : null}
         </div>
         {getInternalPickerSelects()}
       </div>
-      {error}
+      {error && (
+        <div className={`${prefix}--form-requirement`}>
+          {invalid ? invalidText : warningText}
+        </div>
+      )}
```

Comment:
I think we should be more intentional with the variable naming here. looks like this chunk of text shows up both when there's an "invalid" state and a "warning" state, I don't think we should consider a warning state an error, I'd either rename this variable or just change it to `invalid || warning && ....`, same goes for line 305